### ISSUE TITLE
InvalidOperationException "Collection was modified; enumeration operation may not execute." at 10/21/2024 5:54:59 AM -03:00

### ISSUE THREAD

**Issue by mrknify** [njthomson/SrvSurvey#134]
### What happened?

Carrier jumping, while using chrome.
Not the first time i've done this but the first error doing this.

### Version

1.1.3.0

### Exception message

Collection was modified; enumeration operation may not execute.

### Exception stack

   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeList(JsonWriter writer, IEnumerable values, JsonArrayContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonConvert.SerializeObjectInternal(Object value, Type type, JsonSerializer jsonSerializer)
   at Newtonsoft.Json.JsonConvert.SerializeObject(Object value, Type type, Formatting formatting, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.SerializeObject(Object value, Formatting formatting, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.SerializeObject(Object value, Formatting formatting)
   at SrvSurvey.game.Data.Save()
   at SrvSurvey.game.Game.Journals_onJournalEntry(JournalEntry entry, Int32 index)

### Relevant log output

```shell
{"part":1,"Odyssey":true,"gameversion":"4.0.0.1809","build":"r305601/r0 ","timestamp":"2024-10-21T08:14:25Z","event":"Fileheader"}
05:39:48: CargoFile updated: 2 items in: Ship
  drones: 2
05:39:48: Mode change InternalPanel => StationServices
05:39:58: Mode change StationServices => Docked
05:40:11: Mode change Docked => Flying
05:40:44: Mode change Flying => SuperCruising
05:40:46: Mode change SuperCruising => FSS
05:40:46: Program.Creating plotter: PlotFSS
05:40:46: Program.Showing plotter: PlotFSS
05:41:21: Mode change FSS => SuperCruising
05:41:21: Program.Closing plotter: PlotFSS
05:41:23: Mode change SuperCruising => SystemMap
05:41:39: Mode change SystemMap => SuperCruising
05:41:54: Mode change SuperCruising => Flying
05:41:55: Mode change Flying => ExternalPanel
05:42:00: Mode change ExternalPanel => Flying
05:42:05: Mode change Flying => ExternalPanel
05:42:14: Mode change ExternalPanel => Flying
05:43:17: Mode change Flying => Docked
05:43:21: Mode change Docked => StationServices
05:51:10: Mode change StationServices => Docked
```

**Comment by njthomson** [comment]
Thank you @mrknify for reporting this, I must confess FC jumping is not something I get to test very often. I recently released an update to the Windows store and I recommend updating.

I will take a look at the code mentioned above and see what I can do. Thank you again.

**Comment by mrknify** [comment]
No prob. I had this error on SRV Survey as well but didn't report it there. (Same error same jump etc..)

This may be more of a me error, I'll try to duplicate it to get replication for better trouble shooting.

**Comment by njthomson** [comment]
Alas there's not enough information here to diagnose a root cause.

@mrknify - there have been many quality fixes since version 1.1.3.0 - I recommend updating to the latest store build and give me a shout if you see anything like this again. (And if it does happen, could you share the whole log file too?)
